server/src/main/java/org/candlepin/controller/ModeManager.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) 2009 - 2012 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.controller;
+
+import org.candlepin.model.CandlepinModeChange;
+import org.candlepin.model.CandlepinModeChange.Mode;
+
+/**
+ * Class responsible for storing and managing the current Candlepin mode.
+ */
+public interface ModeManager {
+
+    Mode getCurrentMode();
+    CandlepinModeChange getCurrentCandlepinMode();
```

Comment:
I agree its confusing. My suggestion is to remove the getCurrentMode() and I will rename getCurrentCandlepinMode() to getLastCandlepinModeChange()

I will go ahead with this, let me know if you dont agree